python/semantic_kernel/data/vector_search/vector_text_search.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from typing import TYPE_CHECKING, Any, Generic, TypeVar
+
+from semantic_kernel.data.search_options import SearchOptions
+from semantic_kernel.data.text_search.utils import create_options
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+if TYPE_CHECKING:
+    from semantic_kernel.data.kernel_search_results import KernelSearchResults
+    from semantic_kernel.data.vector_search.vector_search_result import VectorSearchResult
+
+TModel = TypeVar("TModel")
+
+logger = logging.getLogger(__name__)
+
+
+@experimental_class
+class VectorTextSearchMixin(Generic[TModel]):
+    """The base class for all text searches."""
+
+    async def text_search(
+        self,
+        search_text: str,
+        options: SearchOptions | None = None,
+        **kwargs: Any,
+    ) -> "KernelSearchResults[VectorSearchResult[TModel]]":
+        """Search the vector store for records that match the given embedding and filter.
+
+        Args:
+            search_text: The query to search for.
+            options: options, should include query_text
+            **kwargs: if options are not set, this is used to create them.
+
+        Raises:
+            VectorSearchOptionsException: raised when the options given are not correct.
+            SearchResultEmptyError: raised when there are no results returned.
+
+        """
+        from semantic_kernel.data.vector_search.vector_search import VectorSearchBase
+
+        if not isinstance(self, VectorSearchBase):
+            raise NotImplementedError("This method can only be used in combination with the VectorSearchBase.")
+        options = create_options(self.options_class, options, **kwargs)
+        return await self._inner_search(search_text=search_text, options=options)
```

Comment:
Question: how will it work if a class implements all vector text search, vectorizable text search and vectorized search? Does it have to switch between the cases by the options?

Reply:
I had a look and it doesn't add much to separate these out, while it makes it easier to implement your own vector store.